backend/chat_messages/views/message_views.py

```diff
@@ -179,40 +172,45 @@ def get_queryset(self):
                     and number_of_participants == 2
                 ):
                     filtered_chats = filtered_chats.exclude(id=chat.id)
-               
+
             return filtered_chats
         else:
             return []
 
 
-
 class GetSearchedChat(ListAPIView):
     permission_classes = [IsAuthenticated]
     serializer_class = MessageParticipantSerializer
-    
+
     pagination_class = ChatsPagination
-    
+
     def get_queryset(self):
-        
-        query = self.request.query_params.get('search')
 
+        query = self.request.query_params.get("search")
+        current_user_full_name = (
+            self.request.user.first_name + " " + self.request.user.last_name
+        )
         chat_ids = Participant.objects.filter(
             user=self.request.user, is_active=True
-        ).values_list('chat', flat=True)
-        
-        participants_matching_query = Participant.objects.annotate(
-            full_name=Concat('user__first_name', V(' '), 'user__last_name')
-        ).filter(
-            chat__in=chat_ids, full_name__icontains=query, is_active=True
+        ).values_list("chat", flat=True)
+
+        participants_matching_query = (
+            Participant.objects.annotate(
+                full_name=Concat("user__first_name", V(" "), "user__last_name"),
+            )
+            .filter(chat__in=chat_ids, full_name__icontains=query, is_active=True)
+            .exclude(
+                chat__in=chat_ids,
+                full_name__icontains=current_user_full_name,
+                is_active=True,
+            )
         )
```

Comment:
You can reduce two queries we make here for Participant and MessageParticipants into one. You can use Prefetch Related: https://docs.djangoproject.com/en/4.1/ref/models/querysets/#prefetch-related